navbar.ts

```diff
@@ -0,0 +1,433 @@
+          }
+          this.overlayContainer[key][0].style.display = "block";
+          this.currentPath.push(anchor);
+        };
+      }
+    }
+
+    const elements = document.querySelectorAll("#close");
+    elements.forEach((element) => {
+      (element as HTMLAnchorElement).onclick = () => this.handleClose();
+    });
+
+    const ele = document.querySelectorAll("#back");
+    ele.forEach((element) => {
+      (element as HTMLAnchorElement).onclick = () => this.handleBack();
+    });
+
+    this.menu.classList.add(this.canvascls);
+  }
+
+  overlayNode(parentName: string = "menu"): HTMLElement {
+    const dive = document.createElement("div");
+    if (parentName == "menu") {
+      const div = document.createElement("div");
+      div.appendChild(this.closeElement);
+      dive.appendChild(div);
+    }
+    this.parentContainer[parentName].forEach((child) => {
+      const div = document.createElement("div");
+      div.appendChild(child);
+      dive.appendChild(div);
+      const attributes = this.parseAttributes(child.getAttribute("attr") || "");
+      this.overlayCreator(attributes.nm);
+    });
+
+    return dive;
+  }
+
+  overlayCreator(anchorName: string): void {
+    const children = this.parentContainer[anchorName];
+
+    const closeCloned = this.closeElement.cloneNode(true) as HTMLAnchorElement;
+    closeCloned.style.display = "block";
+
+    const backCloned = this.backElement.cloneNode(true) as HTMLAnchorElement;
+    if (anchorName == "menu") backCloned.style.display = "none";
+    else {
+      backCloned.style.display = "block";
+    }
+    const anchorE = this.getAnchorElement(anchorName);
+    const overlay = document.createElement("div");
+    if (children) {
+      overlay.id = `overlay-dropdown-${anchorName}`;
+      overlay.style.position = "absolute";
+      overlay.style.top = "0";
+      overlay.style.left = "0";
+      overlay.style.width = "100%";
+      overlay.style.height = "100%";
+      overlay.style.backgroundColor = "rgba(0, 0, 0, 1)";
+      overlay.style.zIndex = "1000";
+      overlay.style.display = "flex";
+      overlay.style.flexDirection = "column";
+
+      const parentName = document.createElement("h3");
+      if (anchorE) parentName.textContent = anchorE.innerHTML;
+      parentName.style.marginRight = "10px";
+
+      const header = document.createElement("div");
+      header.style.padding = "10px";
+      header.style.color = "#ffffff";
+      header.style.display = "flex";
+      header.style.width = "100%";
+
+      header.appendChild(backCloned);
+
+      header.appendChild(parentName);
+      header.appendChild(closeCloned);
+      overlay.appendChild(header);
+
+      const dbox = document.createElement("div");
+
+      const anchor = this.getAnchorElement(anchorName);
+      if (anchor) {
+        const attributes = this.parseAttributes(
+          anchor.getAttribute("attr") || ""
+        );
+        if (attributes.mode == "overlay") {

```

Comment:
If mode is not passed as part of this attributes, we need to use the value passed in the mode property.